### ISSUE TITLE
SplFileInfoRepresentation: TypeError date(): Argument #2 ($timestamp) must be of type ?int, bool given

### ISSUE THREAD

**Issue by kenjis** [kint-php/kint#407]
Kint v5.0.4

See https://github.com/codeigniter4/CodeIgniter4/pull/7337

https://github.com/kint-php/kint/blob/b8b1da88746ab9d27b7454d6720ee95937e31b7a/src/Zval/Representation/SplFileInfoRepresentation.php#L185

`SplFileInfo::getMTime()` may return `false`.
https://www.php.net/manual/en/splfileinfo.getmtime.php

**Comment by jnvsor** [comment]
I ran into this problem with `null` but I was under the impression you _couldn't_ get `false` from `getMTime` without a valid `getRealPath` - I want to add this to the test suite, can you tell me what type of file causes this edge case please?

**Comment by kenjis** [comment]
Yes, we need invalid file uploading.

Get https://github.com/kenjis/ci4-file-upload

```
$ composer install
```

```diff
--- a/app/Controllers/Upload.php
+++ b/app/Controllers/Upload.php
@@ -15,6 +15,9 @@ class Upload extends BaseController
 
     public function upload()
     {
+        $img = $this->request->getFile('userfile');
+        var_dump($img->getMTime()); exit;
+
         $validationRule = [
             'userfile' => [
                 'label' => 'Image File',
```

```
$ php spark serve
```

Navigate to http://localhost:8080/upload, and push "upload" button.
You will see `false`.

**Comment by jnvsor** [comment]
Looks like `getRealPath` will return cwd when called on a `new SplFileInfo("")` - though you should probably handle that edge case in your framework too. Fixed in 1033fc4a5075e15b9b46be20c246069ab44bc243
